src/main/java/com/github/pires/hazelcast/HazelcastDiscoveryController.java

```diff
@@ -119,8 +120,17 @@ public void run(String... args) {
             if (endpoints != null) {
                 if (endpoints.subsets != null && !endpoints.subsets.isEmpty()) {
                     endpoints.subsets.forEach(subset -> {
-                        subset.addresses.forEach(
+                    	if (subset.addresses != null && subset.addresses.isEmpty()) {
+                        	subset.addresses.forEach(
                                 addr -> hazelcastEndpoints.add(addr.ip));
+                        } else {
+                        	// in case of a full cluster restart
+                        	// no address might be ready, in order to allow the cluster
+                        	// to start initially, we will use the not ready addresses
+                        	// as fallback
+                        	subset.notReadyAddresses.forEach(
```

Comment:
You're not validating if `subset.notReadyAddresses` is null and it not empty as you are doing above for `subset.addresses` and you should!